events/source/equeue.c

```diff
@@ -220,15 +226,16 @@ void equeue_dealloc(equeue_t *q, void *p)
         e->dtor(e + 1);
     }
 
-    equeue_mem_dealloc(q, e);
+    if (!IS_USER_ALLOCATED_EVENT(e)) {
+        equeue_mem_dealloc(q, e);
+    }
 }
 
-
 // equeue scheduling functions
 static int equeue_enqueue(equeue_t *q, struct equeue_event *e, unsigned tick)
```

Comment:
It's a problem but we assumed that such situation will not happen, but API allow this.
@kjbracey-arm  to make to add more safety to `UserAllocatedEvent` whe should leave only `bool try_call` API to prevent double posting ?

Reply:
I feel strongly that it should be possible to safely make a second queue call and have it ignored via some API, so there's no need for separate "have I queued" tracking logic in the client - a sigio handler can just do `post()`.

Maybe this usage should be distinguished via `try_post` (return bool/error) versus `post` (return void, assert on fail).

The other plausible behaviour (which might be preferable when we can pass parameters) would be to actually have a second queue attempt replace the first one (updating parameters, but retaining position in queue). That's indistinguishable if you're passing no parameters.